agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocator.java

```diff
@@ -58,14 +64,21 @@ public void searchFileForGitPropertiesAsync(File file, boolean isJarFile) {
 	private void searchFile(File file, boolean isJarFile) {
 		logger.debug("Searching jar file {} for a single git.properties", file);
 		try {
-			T data = dataExtractor.extractData(file, isJarFile);
-			if (data == null) {
-				logger.debug("No git.properties file found in {}", file.toString());
+			List<T> data = dataExtractor.extractData(file, isJarFile, recursiveSearch);
+			if (data.isEmpty()) {
+				logger.debug("No git.properties files found in {}", file.toString());
+				return;
+			}
+			if (data.size() > 1) {
+				logger.debug("Multiple git.properties files found in {}", file.toString() +
+						". Uploading to multiple projects is not possible with the option teamscale-project.");
+
 				return;
 			}
+			T dataEntry = data.get(0);
 
 			if (foundData != null) {
-				if (!foundData.equals(data)) {
+				if (!foundData.equals(dataEntry)) {
 					logger.error(
```

Comment:
This seems not fully consistent with your change from above. You say that `if (data.size() > 1)` we don't look at either of the git.properties files and return instead. Here, we log an error and keep using the first one we found. Also, do you know, in what scenario we can find another one here?

Reply:
When is `data.size() > 1` possible: In every case in which the given jar file or class folder contains multiple git.properties files. E.g. when you have nested jars with git.properties files.

Re inconsistent handling of errors: Yeah, you're right. We can just use the first one we found. 
Also, I assume with "here" in "here, we log an error" you mean some lines below. As we continue after the error, I'd change that to warning. I also changed the debug log for `data.size() > 1` to a warning.